src/entities/Basket/Basket.controller.ts

```diff
@@ -79,43 +80,43 @@ export const addBasket = async (req: Request, res: Response) => {
 };
 
 export const updateBasketStatus = async (req: Request, res: Response) => {
-  const { error } = validateBasketStatus(req.body);
-  if (error) return res.status(BAD_REQUEST).send(error.details[0].message);
-
-  const user = await User.findById(req.user);
-  if (!user) return res.status(BAD_REQUEST).send("The user is not logged in");
-
-  try {
-    const basket = await Basket.findById({ _id: req.params.id })
-      .populate("createdByUserId", "name")
-      .populate("targetUserID", "name");
-
-    if (!basket) {
-      return res.status(NOT_FOUND).send("There is no basket to be updated");
+    
+    const { error } = validateBasketStatus(req.body);
+    if (error) return res.status(BAD_REQUEST).send(error.details[0].message);
+
+    const user = await User.findById(req.user);
+    if (!user)
+    return res.status(BAD_REQUEST).send("The user is not logged in");
+
+    try {
+        const basket = await Basket
+        .findById( { _id: req.params.id })
+        .populate('createdByUserId', 'name')
+        .populate('targetUserID', 'name')
+        
+        if (!basket) {
+            return res.status(NOT_FOUND).send('There is no basket to be updated')
+        }
+
+        if( basket.createdByUserId ) {
+            if (!basket.createdByUserId.equals(user._id) && !basket.targetUserID.equals(user._id)) {
+                return res.status(FORBIDDEN).send("Basket does not belong to the user");
+            }
+        } 
+
+        if (basket.status === req.body.status) {
+            return res.status(OK).send('Basket status did not change (new status is same as current status)');
+        }
+
+        if (getStatusValue(basket.status) > getStatusValue(req.body.status)) {
+            return res.status(OK).send('Incorrect basket status (you cannot return to the previous status)');
+        }
+
+        await Basket.updateOne( { _id: req.params.id }, { status: req.body.status });
```

Comment:
Does it work? I'm asking because there is no $set

Reply:
It works, I think 'set' is the default for updateOne